src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/resource/VisitResource.kt

```diff
@@ -0,0 +1,154 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.resource
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.AmendVisitRequest
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.CancelVisitRequest
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.CreateVisitRequest
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.CreateVisitResponse
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.service.VisitService
+import javax.validation.Valid
+
+@RestController
+@Validated
+@RequestMapping("/visits", produces = [MediaType.APPLICATION_JSON_VALUE])
+class VisitResource(private val visitService: VisitService) {
```

Comment:
I was wondering whether these should be resources of a prisoner?
e.g. it would be 

`/prisoner/offenderNo/{offenderNo}/visit`

So you must supply a prisoner id separately from the body?

Reply:
Ok, guess this makes sense as there could eventually be reference data endpoints or bookingId related ones

Reply:
Actually what would the amend/cancel urls be? This affects whether they can live in the same resource class

Reply:
Done, for now they are PUTs using `/prisoner/offenderNo/{offenderNo}/visit/{visit}` and `/prisoner/offenderNo/{offenderNo}/visit/{visit}/cancel`